ccl/backupccl: TestBackupDBWithViewOnAdjacentDBRange failed

ccl/backupccl.TestBackupDBWithViewOnAdjacentDBRange [failed](https://tanzanite.cluster.engflow.com/invocations/default/ae61b087-8bf2-4b6b-b7ca-d991bab787ef?testReportRun=2&testReportShard=34&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [cc6ca026319024800395293b0fb18f05dd8eb50e](https://github.com/cockroachdb/cockroach/commits/cc6ca026319024800395293b0fb18f05dd8eb50e):


```
=== RUN   TestBackupDBWithViewOnAdjacentDBRange
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange3079515248
    test_log_scope.go:81: use -show-logs to present logs inline
    backup_test.go:10903: error executing query="BACKUP database db into latest in 'userfile:///a' with revision_history;" args=[]: pq: failed to run backup: running distributed backup to export 1 ranges: exporting /Table/112/{1-2}: batch timestamp 1707979459.347250317,1 must be after replica GC threshold 1707979463.785947106,0
    testutils.go:289: no Invalid Descriptors
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange3079515248
--- FAIL: TestBackupDBWithViewOnAdjacentDBRange (6.39s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=34</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupDBWithViewOnAdjacentDBRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36103

ccl/backupccl.TestBackupDBWithViewOnAdjacentDBRange [failed](https://tanzanite.cluster.engflow.com/invocations/default/e0d64c03-bbe8-4af3-b0cb-5866cef06f13?testReportRun=2&testReportShard=34&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [617bf347978dcc0d711399b1a76402d7f88de958](https://github.com/cockroachdb/cockroach/commits/617bf347978dcc0d711399b1a76402d7f88de958):


```
=== RUN   TestBackupDBWithViewOnAdjacentDBRange
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange3650265112
    test_log_scope.go:81: use -show-logs to present logs inline
    backup_test.go:10903: error executing query="BACKUP database db into latest in 'userfile:///a' with revision_history;" args=[]: pq: failed to run backup: running distributed backup to export 1 ranges: exporting /Table/112/{1-2}: batch timestamp 1708068744.116528095,1 must be after replica GC threshold 1708068748.546007194,0
    testutils.go:289: no Invalid Descriptors
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange3650265112
--- FAIL: TestBackupDBWithViewOnAdjacentDBRange (6.47s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=34</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupDBWithViewOnAdjacentDBRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestBackupDBWithViewOnAdjacentDBRange [failed](https://tanzanite.cluster.engflow.com/invocations/default/01563976-1fe2-4191-bba5-22eea9550bd0?testReportRun=2&testReportShard=34&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [0b7ae19e2b94b851ed8812914f57032aab699811](https://github.com/cockroachdb/cockroach/commits/0b7ae19e2b94b851ed8812914f57032aab699811):


```
=== RUN   TestBackupDBWithViewOnAdjacentDBRange
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange1073761015
    test_log_scope.go:81: use -show-logs to present logs inline
    backup_test.go:10910: error executing query="BACKUP database db into latest in 'userfile:///a' with revision_history;" args=[]: pq: failed to run backup: running distributed backup to export 1 ranges: exporting /Table/112/{1-2}: batch timestamp 1708153565.046164229,1 must be after replica GC threshold 1708153569.512091920,0
    testutils.go:289: no Invalid Descriptors
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange1073761015
--- FAIL: TestBackupDBWithViewOnAdjacentDBRange (6.47s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=34</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupDBWithViewOnAdjacentDBRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestBackupDBWithViewOnAdjacentDBRange [failed](https://tanzanite.cluster.engflow.com/invocations/default/d8143f8a-6e9f-4d29-ac14-e1fa6eea69a7?testReportRun=2&testReportShard=34&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [e39dafe6d8c153301ff43ed2b3ed3e13af9ec72a](https://github.com/cockroachdb/cockroach/commits/e39dafe6d8c153301ff43ed2b3ed3e13af9ec72a):


```
=== RUN   TestBackupDBWithViewOnAdjacentDBRange
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange3079360815
    test_log_scope.go:81: use -show-logs to present logs inline
    backup_test.go:10910: error executing query="BACKUP database db into latest in 'userfile:///a' with revision_history;" args=[]: pq: failed to run backup: running distributed backup to export 1 ranges: exporting /Table/112/{1-2}: batch timestamp 1708238568.157408807,1 must be after replica GC threshold 1708238572.622250443,0
    testutils.go:289: no Invalid Descriptors
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange3079360815
--- FAIL: TestBackupDBWithViewOnAdjacentDBRange (6.56s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=34</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupDBWithViewOnAdjacentDBRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestBackupDBWithViewOnAdjacentDBRange [failed](https://tanzanite.cluster.engflow.com/invocations/default/02d84574-2d78-41d8-be8f-86931ea1b806?testReportRun=1&testReportShard=34&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [e39dafe6d8c153301ff43ed2b3ed3e13af9ec72a](https://github.com/cockroachdb/cockroach/commits/e39dafe6d8c153301ff43ed2b3ed3e13af9ec72a):


```
=== RUN   TestBackupDBWithViewOnAdjacentDBRange
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange1916613916
    test_log_scope.go:81: use -show-logs to present logs inline
    backup_test.go:10910: error executing query="BACKUP database db into latest in 'userfile:///a' with revision_history;" args=[]: pq: failed to run backup: running distributed backup to export 1 ranges: exporting /Table/112/{1-2}: batch timestamp 1708321176.267498135,1 must be after replica GC threshold 1708321180.716273264,0
    testutils.go:289: no Invalid Descriptors
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange1916613916
--- FAIL: TestBackupDBWithViewOnAdjacentDBRange (6.56s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=34</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupDBWithViewOnAdjacentDBRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestBackupDBWithViewOnAdjacentDBRange [failed](https://tanzanite.cluster.engflow.com/invocations/default/7f1d73c5-5ddd-4855-bf60-ae0b8b268601?testReportRun=1&testReportShard=34&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [e50b0ec4d3a53f81e26f3776ae3f3be55d435a9a](https://github.com/cockroachdb/cockroach/commits/e50b0ec4d3a53f81e26f3776ae3f3be55d435a9a):


```
=== RUN   TestBackupDBWithViewOnAdjacentDBRange
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange3235530297
    test_log_scope.go:81: use -show-logs to present logs inline
    backup_test.go:10910: error executing query="BACKUP database db into latest in 'userfile:///a' with revision_history;" args=[]: pq: failed to run backup: running distributed backup to export 1 ranges: exporting /Table/112/{1-2}: batch timestamp 1708499480.449653300,1 must be after replica GC threshold 1708499484.925285394,0
    testutils.go:289: no Invalid Descriptors
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange3235530297
--- FAIL: TestBackupDBWithViewOnAdjacentDBRange (6.51s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=34</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupDBWithViewOnAdjacentDBRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestBackupDBWithViewOnAdjacentDBRange [failed](https://tanzanite.cluster.engflow.com/invocations/default/fe2f43e7-4d0f-4d99-8602-fa507e042fb9?testReportRun=1&testReportShard=34&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [04f0416d526a43741d22fd03966758dcccdeb79f](https://github.com/cockroachdb/cockroach/commits/04f0416d526a43741d22fd03966758dcccdeb79f):


```
=== RUN   TestBackupDBWithViewOnAdjacentDBRange
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange761219499
    test_log_scope.go:81: use -show-logs to present logs inline
    backup_test.go:10910: error executing query="BACKUP database db into latest in 'userfile:///a' with revision_history;" args=[]: pq: failed to run backup: running distributed backup to export 1 ranges: exporting /Table/112/{1-2}: batch timestamp 1708603131.159795003,1 must be after replica GC threshold 1708603135.669694514,0
    testutils.go:289: no Invalid Descriptors
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange761219499
--- FAIL: TestBackupDBWithViewOnAdjacentDBRange (6.59s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=34</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupDBWithViewOnAdjacentDBRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestBackupDBWithViewOnAdjacentDBRange [failed](https://tanzanite.cluster.engflow.com/invocations/default/90411a40-d13d-44c1-bee6-2218265e598c?testReportRun=2&testReportShard=34&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [6d65201b9b603e0b3fcf1d509ec23edfdd68de45](https://github.com/cockroachdb/cockroach/commits/6d65201b9b603e0b3fcf1d509ec23edfdd68de45):


```
=== RUN   TestBackupDBWithViewOnAdjacentDBRange
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange1028626557
    test_log_scope.go:81: use -show-logs to present logs inline
    backup_test.go:10910: error executing query="BACKUP database db into latest in 'userfile:///a' with revision_history;" args=[]: pq: failed to run backup: running distributed backup to export 1 ranges: exporting /Table/112/{1-2}: batch timestamp 1708676922.608009916,1 must be after replica GC threshold 1708676927.095543001,0
    testutils.go:289: no Invalid Descriptors
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupDBWithViewOnAdjacentDBRange1028626557
--- FAIL: TestBackupDBWithViewOnAdjacentDBRange (6.41s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=34</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupDBWithViewOnAdjacentDBRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/backupccl/utils_test.go
+++ b/pkg/ccl/backupccl/utils_test.go
@@ -287,11 +287,15 @@ func uriFmtStringAndArgs(uris []string, startIndex int) (string, []interface{}) 
 // span.
 func waitForTableSplit(t *testing.T, conn *gosql.DB, tableName, dbName string) {
 	t.Helper()
+	query := fmt.Sprintf(`SELECT count(*)
+  FROM crdb_internal.ranges
+ WHERE start_key = crdb_internal.table_span('%s.%s'::regclass::oid::int)[1]`,
+		tree.NameString(dbName),
+		tree.NameString(tableName))
+
 	testutils.SucceedsSoon(t, func() error {
 		count := 0
-		if err := conn.QueryRow(
-			fmt.Sprintf("SELECT count(*) FROM [SHOW RANGES FROM TABLE %s.%s]",
-				tree.NameString(dbName), tree.NameString(tableName))).Scan(&count); err != nil {
+		if err := conn.QueryRow(query).Scan(&count); err != nil {
 			return err
 		}
 		if count == 0 {

